### PR TITLE
Add read threshold for rnafusion

### PR DESCRIPTION
**Pull request by fevac**
## Description

Fixes https://github.com/Clinical-Genomics/cg/issues/2545

### Fixed

- Automatic start of rnafusion by adding read threshold

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b nf_treshold_reads -a
    ```

### How to test

- [x] Set cuddlyhen and politeglider as Action empty in StatusDB stage and Reads Updated At  with a date later than any completed analysis (Analysis.completed_at < Sample.reads_updated_at) and created analysis (Analysis.created_at < Sample.reads_updated_at). 
- [x] Run `cg workflow rnafusion start-available -d` from the master branch -> make sure that both cases are picked up
- [x]  Run `cg workflow rnafusion start-available -d` from this branch -> make sure that only politeglider is picket up

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [x] Tests executed by EFC
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Review by seallard — APPROVED** [review]
👍  Seems like some test is breaking

File: cg/meta/workflow/nf_analysis.py [hunk 1]
```diff
@@ -56,6 +56,12 @@ def process(self):
     def process(self, process: Process):
         self._process = process
 
+    @property
+    def threshold_reads(self):
```
Comment (ChrOertlin) [inline cg/meta/workflow/nf_analysis.py:60]:
```suggestion
    def pass_reads_threshold(self) -> bool:
```

**Review by ChrOertlin — APPROVED** [review]
Looks good, just a suggestion to make the intention clearer

File: cg/meta/workflow/nf_analysis.py (hunk 1 above)
Comment (fevac) [reply cg/meta/workflow/nf_analysis.py:60]: I will not do that here. This is the name required by analysis api and the name used in all other pipelines

Comment (henrikstranneheim) [inline cg/meta/workflow/nf_analysis.py:60]:
```suggestion
    def use_read_count_threshold(self):
```

**Review by henrikstranneheim — APPROVED** [review]
👍 . Some test needs fixing

File: cg/meta/workflow/nf_analysis.py (hunk 1 above)
Comment (fevac) [reply cg/meta/workflow/nf_analysis.py]: well. ended up doing it here as Henrik also had the same comment

Comment (fevac) [reply cg/meta/workflow/nf_analysis.py]: done and fixed pytests

File: cg/meta/workflow/nf_analysis.py [hunk 2]
```diff
@@ -56,6 +56,12 @@ def process(self):
     def process(self, process: Process):
         self._process = process
 
+    @property
+    def use_read_count_threshold(self):
```
Comment (henrikstranneheim) [inline cg/meta/workflow/nf_analysis.py:60]:
```suggestion
    def use_read_count_threshold(self) -> bool:
```

File: tests/conftest.py [hunk 3]
```diff
@@ -2773,6 +2770,7 @@ def rnafusion_context(
         status_db,
         internal_id=sample_id,
         reads_updated_at=datetime.now(),
+        reads=200000000,
```
Comment (henrikstranneheim) [inline tests/conftest.py:2773]:
```suggestion
        reads=200_000_000,
```
Perhaps make a fixture instead of hard coding?

**Review by henrikstranneheim — APPROVED** [review]
Great - I think the renaming improves readability a lot 👍

File: cg/meta/workflow/nf_analysis.py (hunk 2 above)
Comment (fevac) [reply cg/meta/workflow/nf_analysis.py:60]: done

**Comment by fevac** [comment]
Tests are failing. Need to figure out what's up

**Comment by fevac** [comment]
- Deployed to stage
image


- Deployed to prod
image

**Pull request merged**